### PR TITLE
Custom JS Plugin

### PR DESCRIPTION
**Pull request by JokerJosh1234**
allows the user to input code into plugin settings, then that code will execute every time they launch discord. theres also a button to check for errors. theres also error catching when discord starts for custom js.

### CODE REVIEW

**Comment by rozbrajaczpoziomow** [comment]
https://github.com/Vendicated/Vencord/pull/2420
me when

**Comment by Vendicated** [comment]
yeah this is a duplicate of abovementioned pr, so i'll be closing this in favour of that pr

thank you for your effort regardless!

**Comment by JokerJosh1234** [comment]
isnt that one for commands?

File: src/plugins/customJs/index.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+                    Notifications.showNotification({
+                        title: "Custom JS",
+                        body: (err as Error).toString(),
+                    });
+                }
+            };
+
+            return (
+                <div>
+                    <h3 className="h3__1f119 title__1b7e8 defaultColor_e04945 defaultMarginh3_a8638d">Custom JS:</h3>
```
Comment (Sqaaakoi) [inline src/plugins/customJs/index.tsx:53]: don't hardcode these names like this. Use the provided React components that generate things like this; look at other plugins for examples

File: src/plugins/customJs/index.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                    <div className="inputWrapper_ecb776">
```
Comment (Sqaaakoi) [inline src/plugins/customJs/index.tsx:54]: There's a styled TextArea component for this too

File: src/plugins/customJs/index.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                        <textarea
+                            className="inputDefault__22335 input_f27786"
+                            placeholder="console.log('Hello World!');"
+                            style={{ minHeight: "100px", width: "100%", resize: "vertical" }}
+                            onChange={handleChange}
+                            onKeyUp={handleChange}
+                            defaultValue={customJs}
+                        />
+                    </div>
+                    <button type="button" className="button__581d0 lookFilled__950dd colorBrand__27d57 sizeSmall_da7d10 grow__4c8a4" onClick={compileCode} style={{ marginTop: "10px" }}>
```
Comment (Sqaaakoi) [inline src/plugins/customJs/index.tsx:64]: Button component also exists

File: src/plugins/customJs/index.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+export const settings = definePluginSettings({
+    customJs: {
+        description: "Custom JS",
+        type: OptionType.COMPONENT,
+        component: () => {
+            const [customJs, setCustomJs] = useState("");
+
+            useEffect(() => {
+                (async () => {
+                    const storedJs = await DataStore.get("CustomJS_js");
```
Comment (Sqaaakoi) [inline src/plugins/customJs/index.tsx:23]: why are you using `DataStore` within a settings component? use `settings.store.customJs` to refer to the stored setting

**Comment by JokerJosh1234** [comment]
yeah that one looks like it registers slash commands, this is just running custom javascript in general

**Review by Sqaaakoi — COMMENTED** [review]
Like others have said; such an idea has been implemented before; and knowing Vencord users, such an idea isn't particularly safe to ship.
I'm just leaving some feedback based on what you have written to improve on building future plugins.

File: src/plugins/customJs/index.tsx (hunk 4 above)
Comment (JokerJosh1234) [reply src/plugins/customJs/index.tsx:23]: if i fix all the suggested changes, would the plugin be considered?

**Comment by Vendicated** [comment]
image

**Pull request closed without merging**